OpenRA.Mods.Common/ActorExts.cs

```diff
@@ -141,6 +141,10 @@ public static void NotifyBlocker(this Actor self, IEnumerable<CPos> positions)
 					return false;
 			}
 
+			// Occupied by any actor, including ally/friendly units
+			if (self.World.ActorMap.GetActorsAt(pos).Any())
```

Comment:
This isn't *quite* correct: there's no reason why a harvester shouldn't be able to squash an infantry standing on a resource cell and then harvest it.  It would be better if you could pass `MobileInfo` into this method and then query `CanEnterCell` (note: `Harvester` already requires `Mobile` so we aren't losing any generality) instead.